torchgeo/datasets/urban3d.py

```diff
@@ -0,0 +1,281 @@
+                    dsm=os.path.join(image_root, f"{prefix}_DSM.tif"),
+                    binary_mask=os.path.join(target_root, f"{prefix}_GTL.tif"),
+                    instance_mask=os.path.join(target_root, f"{prefix}_GTI.tif"),
+                )
+            )
+        return files
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if the dataset is not found
+        """
+        # Check if the files already exist
+        exists = []
+        for directory in self.directories.values():
+            exists.append(os.path.exists(os.path.join(self.root, directory)))
+        if all(exists):
+            return
+
+        raise RuntimeError(
+            "Dataset not found in `root` directory, "
+            "specify a different `root` directory."
+        )
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+        alpha: float = 0.5,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample return by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional suptitle to use for figure
+            alpha: opacity with which to render predictions on top of the imagery
+
+        Returns;
+            a matplotlib Figure with the rendered sample
+        """
+        ncols = 5
+        image = sample["image"][:3].permute(1, 2, 0).numpy()
+        image = percentile_normalization(image, axis=(0, 1))
+        image = (image * 255).astype(np.uint8)
+        dtm = percentile_normalization(sample["image"][3].numpy(), lower=0, upper=99)
+        dsm = percentile_normalization(sample["image"][4].numpy(), lower=0, upper=99)
+        ndsm = percentile_normalization(sample["image"][5].numpy(), lower=0, upper=99)
+        tensor = torch.from_numpy(image).permute(2, 0, 1)  # type: ignore[attr-defined]
+        mask = draw_segmentation_masks(
+            image=tensor,
+            masks=sample["mask"].to(torch.bool),  # type: ignore[attr-defined]
+            alpha=alpha,
+            colors="red",
+        ).permute(1, 2, 0)
+
+        showing_predictions = "prediction" in sample
+        if showing_predictions:
+            preds = draw_segmentation_masks(
+                image=tensor,
+                masks=sample["prediction"].to(torch.bool),  # type: ignore[attr-defined]
+                alpha=alpha,
+                colors="red",
+            ).permute(1, 2, 0)
+            ncols += 1
+
+        fig, axs = plt.subplots(nrows=1, ncols=ncols, figsize=(10, ncols * 10))
+        axs[0].imshow(image)
+        axs[0].axis("off")
+        axs[1].imshow(dtm)
+        axs[1].axis("off")
+        axs[2].imshow(dsm)
+        axs[2].axis("off")
+        axs[3].imshow(ndsm)
+        axs[3].axis("off")
+        axs[4].imshow(mask)
+        axs[4].axis("off")
```

Comment:
This could be a for-loop, at least the axis off part. Just a suggestion, not required